diff --git a/external/CMakeLists.txt b/external/CMakeLists.txt
index c2ad2d4a..3dc49273 100644
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -76,6 +76,7 @@ set(EXTERNAL_INCLUDES "")
 # ==========
 
 if(${CHFL_SYSTEM_ZLIB})
+    message(WARNING "64-bit offsets are often not supported by the system zlib, consider using CHFL_SYSTEM_ZLIB=OFF")
     find_package(ZLIB REQUIRED)
     set(ZLIB_LIBRARIES ${ZLIB_LIBRARIES} PARENT_SCOPE)
     set(ZLIB_OBJECTS "" PARENT_SCOPE)
@@ -93,6 +94,7 @@ list(APPEND EXTERNAL_INCLUDES ${ZLIB_INCLUDE_DIRS})
 # ==========
 if(${CHFL_SYSTEM_LZMA})
     find_package(LibLZMA REQUIRED)
+    set(LIBLZMA_LIBRARY ${LIBLZMA_LIBRARY} PARENT_SCOPE)
     set(LZMA_OBJECTS "" PARENT_SCOPE)
 else()
     external_library(lzma)
@@ -108,6 +110,7 @@ list(APPEND EXTERNAL_INCLUDES ${LIBLZMA_INCLUDE_DIRS})
 # ==========
 if(${CHFL_SYSTEM_BZIP2})
     find_package(BZip2 REQUIRED)
+    set(BZIP2_LIBRARIES ${BZIP2_LIBRARIES} PARENT_SCOPE)
     set(BZIP2_OBJECTS "" PARENT_SCOPE)
 else()
     external_library(bzip2)
